src/components/VisualizationStep.tsx

```diff
@@ -33,7 +34,13 @@ const VisualizationStep = ({ data }: NodeProps<IVizStepNodeData>) => {
 
   const { addAlert } = useAlert() || {};
 
-  const onMiniCatalogClickAdd = (selectedStep: IStepProps) => appendStep(selectedStep);
+  const onMiniCatalogClickAdd = (selectedStep: IStepProps) => {
```

Comment:
This might end up conflicting with Iram's PR, I will have to check

Reply:
well that's about merge order .. if you merge this later I will rebase and fix conflicts, otherwise it's up to Mohammadi :)